### PR TITLE
Add new custom HTTP mechanism quickstart

### PR DESCRIPTION
**Pull request by sgilda**
@ctomc: This was adapted from @darranl  blog: http://darranl.blogspot.com/

I used his source, which consisted of multiple projects and packaged so that is could all be done from the one `http-custom-mechanism` quickstart. I also changed the instructions a bit to flow with the steps you take to deploy, test, configure, retest.

I do need to add some details about the classes in the README file, but the basics are there. Let me know what you think.

### CODE REVIEW

**Comment by sgilda** [comment]
@ctomc : Hold off on this one. I might package the web app differently.

**Comment by ctomc** [comment]
Ok

**Comment by sgilda** [comment]
Quickstart is updated. It needs a thorough review.

File: http-custom-mechanism/pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        <dependency>
+            <groupId>org.jboss.spec.javax.servlet</groupId>
+            <artifactId>jboss-servlet-api_3.1_spec</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>3.8.1</version>
```
Comment (ctomc) [inline http-custom-mechanism/pom.xml]: remove version, it is managed in parent

File: http-custom-mechanism/webapp/pom.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        <dependency>
+            <groupId>org.jboss.spec.javax.servlet</groupId>
+            <artifactId>jboss-servlet-api_3.1_spec</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>3.8.1</version>
```
Comment (ctomc) [inline http-custom-mechanism/webapp/pom.xml]: remove

**Comment by ctomc** [comment]
@bmaxwell do you think having QS like this is something users & CU are in need of?

File: http-custom-mechanism/pom.xml (hunk 1 above)
Comment (sgilda) [reply http-custom-mechanism/pom.xml]: Removed.

File: http-custom-mechanism/webapp/pom.xml (hunk 2 above)
Comment (sgilda) [reply http-custom-mechanism/webapp/pom.xml]: Removed.

**Comment by sgilda** [comment]
@ctomc : Feel free to close this one if you don't feel it is useful.

**Pull request merged**